packages/network-controller/src/rpc-service/rpc-service.ts

```diff
@@ -485,26 +488,43 @@ export class RpcService implements AbstractRpcService {
     } catch (error) {
       if (error instanceof HttpError) {
         const status = error.httpStatus;
-        if (status === 405) {
-          throw rpcErrors.methodNotFound();
+        if (status === 401) {
+          throw new JsonRpcError(-33100, 'Unauthorized.', {
+            httpStatus: status,
+          });
+        }
+        if (status === 501) {
```

Comment:
Hmm... `methodNotFound` means that the JSON-RPC method was unrecognized and could not be "executed", but 501 means that the _HTTP method_ was unrecognized. That is a server error but could also be a client error too (if they fat-fingered the HTTP method for instance).

Maybe we should not handle 501 and have the catch-all cover this?

Reply:
done here https://github.com/MetaMask/core/pull/5843/commits/e502671c021ee2f775c1d033ff0f93551acf38fb